test/plog/LogStreamTest.cpp

```diff
@@ -0,0 +1,449 @@
+
+                auto request = dto::PersistenceClusterCreateRequest{.cluster=std::move(cluster1)};
+                return RPC().callRPC<dto::PersistenceClusterCreateRequest, dto::PersistenceClusterCreateResponse>(dto::Verbs::CPO_PERSISTENCE_CLUSTER_CREATE, request, *_cpoEndpoint, 1s);
+            })
+            .then([this] (auto&& response) {
+                // Init and create a logstream
+                auto& [status, resp] = response;
+                K2EXPECT(log::ltest, status, Statuses::S201_Created);
+
+                ConfigVar<String> configEp("cpo_url");
+                String cpo_url = configEp();
+
+                return _mmgr.init(cpo_url, "Partition-1", "Persistence_Cluster_1", false);
+            })
+            .then([this] { return runTest1();})
+            .then([this] { return runTest2();})
+            .then([this] { return runTest3();})
+            .then([this] {
+                K2LOG_I(log::ltest, "======= All tests passed ========");
+                exitcode = 0;
+            })
+            .handle_exception([this](auto exc) {
+                try {
+                    std::rethrow_exception(exc);
+                } catch (RPCDispatcher::RequestTimeoutException& exc) {
+                    K2LOG_E(log::ltest, "======= Test failed due to timeout ========");
+                    exitcode = -1;
+                } catch (std::exception& e) {
+                    K2LOG_E(log::ltest, "======= Test failed with exception [{}] ========", e.what());
+                    exitcode = -1;
+                }
+            })
+            .finally([this] {
+                K2LOG_I(log::ltest, "======= Test ended ========");
+                seastar::engine().exit(exitcode);
+            });
+        });
+        _testTimer.arm(0ms);
+        return seastar::make_ready_future<>();
+    }
+
+    seastar::future<> runTest1() {
+        K2LOG_I(log::ltest, ">>> Test1: Write and Read a Value from/to one log stream");
+        _logStream = _mmgr.obtainLogStream(LogStreamType::WAL);
+        String header("0", 10000);
+        Payload payload([] { return Binary(20000); });
+        payload.write(header);
+        // write a string to the first log stream
+        return _logStream->append(std::move(payload))
+        .then([this, header] (auto&& response){
+            auto& [status, plogId, appended_offset] = response;
+            if (!status.is2xxOK()){
+                throw std::runtime_error("cannot append data!");
+            }
+            _initPlogId = plogId;
+            return _logStream->read(plogId, 0, appended_offset);
+        })
+        .then([this, header] (auto&& response){
+            auto& [status, continuation_token, payload] = response;
+            if (!status.is2xxOK()){
+                throw std::runtime_error("cannot read data!");
+            }
+            K2EXPECT(log::ltest, payload.getSize(), 10005);
+            String str;
+            payload.seek(0);
+            payload.read(str);
```

Comment:
validate here that you have a string of size 10000 and it contains only the character '0'

Reply:
I think K2EXPECT(log::ltest, str, header) verified this?